badgecheck/verifier.py

```diff
@@ -41,12 +42,20 @@ def call_task(task_func, task_meta, store):
 def verify(badge_input):
     """
     Verify and validate Open Badges
-    :param badge_input: str (url or json)
+    :param badge_input: str (url or json) or python file-like object (baked badge image)
     :return: dict
     """
     store = create_store(main_reducer, INITIAL_STATE)
 
-    store.dispatch(store_input(badge_input))
+    if hasattr(badge_input, 'read') and hasattr(badge_input, 'seek'):
+        badge_input.seek(0)
+        badge_data = unbake(badge_input)
+        if not badge_data:
+            raise ValueError("Files as badge input must be baked images.")
+    else:
+        badge_data = badge_input
+
+    store.dispatch(store_input(badge_data))
```

Comment:
It feels safer to put the results of unbaking into a new var, and then do the copying into that var for all things that aren't file-like, rather than write back over the passed in variable... I don't know if that's being naive or overly paranoid, though.